test_unittests_server.py

```diff
@@ -0,0 +1,76 @@
+# -*- coding: utf-8 -*-
+from __future__ import unicode_literals
+import server
+import pytest
+
+
+################
+# UNIT TESTS
+################
+
+
+def test_response_ok():
+    response = server.response_ok(b"123456789", "text/plain")
+    assert "text/plain" in response
+    assert '46' in response
+
+
+def test_response_error():
+    response = server.response_error(
+        b"405",
+        b"Method Not Allowed",
+        b"GET method required."
+    )
+    assert b"GET method required" in response
+    assert b"text/html" in response
+    assert b"</body>" in response
+
+
+def test_parse_request_rejects_non_get():
+    with pytest.raises(NotImplementedError):
+        server.parse_request(
+            b"POST / HTTP/1.1\r\n"
+            b"Host: http://example.com\r\n"
+            b"\r\n"
+        )
+
+
+def test_parse_request_rejects_non_http_1_1():
+    with pytest.raises(Exception):
```

Comment:
seems to me that your implementation of `parse_request` raises a specific type of exception, not just any exception.  Is there a reason you aren't testing for that specific exception type?